lib/network/modules/components/edges/util/EndPoints.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * Location of all the endpoint drawing routines.
+ *
+ * Every endpoint has its own drawing routine, which contains an endpoint definition.
+ *
+ * The endpoint definitions must have the following properies:
+ *
+ * - (0,0) is the connection point to the node it attaches to
+ * - The endpoints are orientated to the positive x-direction
+ * - The length of the endpoint is at most 1
+ *
+ * As long as the endpoint classes remain simple and not too numerous, they will be contained within this module.
+ * All classes here except `EndPoints` should be considered as private to this module.
+ */
+
+// NOTE: When a typedef is isolated in a separate comment block, an actual description is generated for it,
+//       using the rest of the commenting in the code block. Usage of typedef in other comments then
+//       link to there. TIL.
+//
+//       Also noteworthy, all typedef's set up in this manner are collected in a single, global page 'global.html'.
+//       In other words, it doesn't matter *where* the typedef's are defined in the code.
+//
+//
+// TODO: add descriptive commenting to given typedef's
+
+/**
+ * @typedef {{type:string, point:Point, angle:number, length:number}} ArrowData
+ *
+ * Object containing instantiation data for a given endpoint.
+ */
+
+/**
+ * @typedef {{x:number, y:number}} Point
+ * 
+ * A point in view-coordinates.
+ */
+
+/**
+ * Common methods for endpoints
+ *
+ * @class
+ */
+class EndPoint {
+
+  /**
+   * Apply transformation on points for display.
+   *
+   * The following is done:
+   * - rotate by the specified angle
+   * - multiply the (normalized) coordinates by the passed length
+   * - offset by the target coordinates
+   *
+   * @param {Array<Point>} points
+   * @param {ArrowData} arrowData
+   * @static
+   */
+  static transform(points, arrowData) {
```

Comment:
static won't work on browsers that don't support ES6

Reply:
Well, we *do* use babel to convert `es2015 syntax` to 'regular' javascript. It's this line in the `package.json`:

```
    "babel-preset-es2015": "^6.24.1",
```

All `es2015` (aka `ES6`) syntax is transpiled to 'regular' javascript upon compilation with `gulp`. To give an example:

```js
class Node {
...
  static parseOptions(parentOptions, newOptions, allowDeletion = false, globalOptions = {}) {
...
  }
};
```

... becomes in the target `vis.js`:

```js
  (0, _createClass3['default'])(Node, [{
...
  }], [{
    key: 'parseOptions',
    value: function parseOptions(parentOptions, newOptions) {
      var allowDeletion = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : false;
      var globalOptions = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : {};
...
  }]);
  return Node;
}();
```

`static` just plain disappears in the target file `vis.js` We've been done the transpilation for some time now. I really think we're OK here. 

----

...if I'm missing something, please enlighten me.

Reply:
Ok, you're right again =)